cider/exec/template/StringOpsIR.cpp

```diff
@@ -342,15 +342,52 @@ std::unique_ptr<CodegenColValues> CodeGenerator::codegenStringOpExpr(
   const int64_t string_ops_handle = reinterpret_cast<int64_t>(string_ops);
   auto string_ops_handle_lv = cgen_state_->llInt(string_ops_handle);
   const auto& return_ti = expr->get_type_info();
-  if (!return_ti.is_string()) {
-    CIDER_THROW(
-        CiderCompileException,
-        "For string op, non-string type return values is not supported currently.");
-  }
 
   const int64_t cider_string_hasher_handle =
       reinterpret_cast<int64_t>(executor()->getCiderStringHasherHandle());
   auto cider_string_hasher_handle_lv = cgen_state_->llInt(cider_string_hasher_handle);
+  if (!return_ti.is_string()) {
```

Comment:
dedup the code with the following part? https://github.com/intel/BDTK/blob/7f18c39f4cddaee713e0c4017081a3f70b0cddb0/cider/exec/template/StringOpsIR.cpp#L272-L302